distsys/resources/incrementalmap.go

```diff
@@ -0,0 +1,124 @@
+package resources
+
+import (
+	"github.com/UBC-NSS/pgo/distsys"
+	"github.com/benbjohnson/immutable"
+)
+
+// A generic map resource, with hooks to programmatically realize child resources during execution
+// ----------------------------------------------------------------------------------------------------------------
+
+type IncrementalArchetypeMapResource struct {
+	distsys.ArchetypeResourceMapMixin
+	realizedMap  *immutable.Map
```

Comment:
That's because the keys are of type `TLAValue`, which is not natively hashable. We need a map type which handles a custom hasher, which `immutable.Map` does, because it's also in use in `TLAValue`'s implementation.